### PR TITLE
many small updates to the new sd package

### PR DESCRIPTION
**Pull request by brandondutra**
* analyze: error message if --bigquery-table used without --cloud
* analyze prints what it is doing
* analyze, transform, and training can take more than one csv file by passing multiple "--csv-file-pattern's"
* transform has an async flag that does not block until the job is done.

### CODE REVIEW

File: solutionbox/code_free_ml/mltoolbox/code_free_ml/analyze.py [hunk 1]
```diff
@@ -149,11 +150,15 @@ def parse_arguments(argv):
...
       raise ValueError('--output-dir must point to a location on GCS')
-    if args.csv_file_pattern and not args.csv_file_pattern.startswith('gs://'):
+    if (args.csv_file_pattern and
+       not all(x.startswith('gs://') for x in args.csv_file_pattern)):
       raise ValueError('--csv-file-pattern must point to a location on GCS')
     if args.csv_schema_file and not args.csv_schema_file.startswith('gs://'):
       raise ValueError('--csv-schema-file must point to a location on GCS')
 
+  if not args.cloud and args.bigquery_table:
+    raise ValueError('--cloud must be used with --bigquery-table')
```
Comment (qimingj) [inline solutionbox/code_free_ml/mltoolbox/code_free_ml/analyze.py]: Should it be "--bigquery-table must be used with --cloud"?

File: solutionbox/code_free_ml/mltoolbox/code_free_ml/analyze.py [hunk 2]
```diff
@@ -218,6 +223,7 @@ def _execute_sql(sql, table):
 
   numerical_vocab_stats = {}
   for col_name, transform_set in six.iteritems(inverted_features_target):
+    sys.stdout.write('Analyzing column %s...' % col_name)
```
Comment (qimingj) [inline solutionbox/code_free_ml/mltoolbox/code_free_ml/analyze.py]: Can we also sys.stdout.flush? Without doing so the message may not appear for a long time.

File: solutionbox/code_free_ml/mltoolbox/code_free_ml/transform.py [hunk 3]
```diff
@@ -494,10 +504,12 @@ def main(argv=None):
...
 
-  with beam.Pipeline(pipeline_name, options=pipeline_options) as p:
-    preprocess(
-        pipeline=p,
-        args=args)
+  p = beam.Pipeline(pipeline_name, options=pipeline_options)
+  preprocess(pipeline=p, args=args)
+  pipeline_result = p.run()
+
+  if not args.async:
```
Comment (qimingj) [inline solutionbox/code_free_ml/mltoolbox/code_free_ml/transform.py]: if async, can we print out a job link?

File: solutionbox/code_free_ml/mltoolbox/code_free_ml/analyze.py (hunk 1 above)
Comment (brandondutra) [inline solutionbox/code_free_ml/mltoolbox/code_free_ml/analyze.py]: done

File: solutionbox/code_free_ml/mltoolbox/code_free_ml/analyze.py (hunk 2 above)
Comment (brandondutra) [inline solutionbox/code_free_ml/mltoolbox/code_free_ml/analyze.py]: done

File: solutionbox/code_free_ml/mltoolbox/code_free_ml/transform.py (hunk 3 above)
Comment (brandondutra) [inline solutionbox/code_free_ml/mltoolbox/code_free_ml/transform.py]: done

**Pull request merged**